samples/net/lldp/src/main.c

```diff
@@ -188,8 +195,19 @@ static enum net_verdict parse_lldp(struct net_if *iface, struct net_pkt *pkt)
 
 static int init_app(void)
 {
-	if (init_vlan() < 0) {
-		LOG_ERR("Cannot setup VLAN");
+	enum ethernet_hw_caps caps;
+	int ret;
+
+	ret = init_vlan();
+	if (ret < 0) {
+		LOG_ERR("Cannot setup VLAN (%d)", ret);
+	}
```

Comment:
So if VLANs are not created, < 0 is returned, and it is enough to log an error? Even without VLANs verything is fine and the sample can work?

Reply:
The VLANs are just optional here, the LLDP will work just fine without it. I just noticed that if VLANs are not supported by the eth driver (like in qemu_x86 and e1000), then there was a crash.

Reply:
Maybe only a warning could be logged instead in that case? In the code it looks like it was meant not to work, and the code reviewer is left wondering what was supposed to happen. The crash does not sound like a fun thing to encounter, either.

Reply:
Sure, we can print warning. There is no crash after this commit.